src/l500/l500-color.cpp

```diff
@@ -493,10 +493,44 @@ namespace librealsense
         return tags;
     }
 
-    std::vector<uint8_t> l500_color::get_raw_intrinsics_table() const
+    ivcam2::intrinsic_rgb l500_color::get_intrinsics_table() const
     {
-        AC_LOG( DEBUG, "RGB_INTRINSIC_GET" );
-        return _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+        // Get RAW data from firmware
+        std::vector< uint8_t > response_vec = _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+
+        if (response_vec.size() == 0)
+            throw invalid_value_exception("Calibration data invalid,buffer size is zero");
+
+        auto resolutions_rgb_table_ptr = reinterpret_cast<const ivcam2::intrinsic_rgb *>(response_vec.data());
+
+        // Get full maximum size of the resolution array and deduct the unused resolutions size from it
+        size_t expected_size = sizeof( ivcam2::intrinsic_rgb )
+                                - ( ( MAX_NUM_OF_RGB_RESOLUTIONS
+                                - resolutions_rgb_table_ptr->resolution.num_of_resolutions )
+                                * sizeof( pinhole_camera_model ) );
+
+        // Validate table size
+        if (expected_size > response_vec.size())
+        {
+            throw invalid_value_exception(
+                to_string() << "Calibration data invalid, buffer too small : expected "
+                << expected_size << " , actual: " << response_vec.size());
+        }
+
+        // Set a new memory allocated intrinsics struct (Full size 5 resolutions)
+        // Copy the relevant data from the dynamic resolution received from the FW
+        ivcam2::intrinsic_rgb  resolutions_rgb_table_output;
+        resolutions_rgb_table_output.common = resolutions_rgb_table_ptr->common;
```

Comment:
I don't like this manual copying... why not memcpy?

Reply:
I will change to librealsense::copy